nixos/modules/services/web-apps/immich.md

```diff
@@ -0,0 +1,265 @@
+
+Example:
+
+```console
+# tar --create --zstd --file /var/backup/immich-library.tar.zst $UPLOAD_LOCATION
+```
+
+Next you should create a backup of your existing database. While this
+guide is not going to modify the original database it might still be
+advisable in case something goes wrong.
+
+Example, assuming your Immich database is running in a container
+named `immich_postgres`:
+
+```console
+# docker exec immich_postgres pg_dump --username immich immich -f /tmp/immich.sql
+# docker cp immich_postgres:/tmp/immich.sql /var/backup/immich.sql
+```
+
+It is advised to store these backups in a safe location, at least
+until you have confirmed that your new installation works.
+
+### Configure Immich NixOS module {#module-immich-migrating-from-docker-compose-configure}
+
+You can now enable Immich in your NixOS configuration as described in
+[Configuring](#module-immich-configuring).
+
+After building and switching to your new configuration, you should
+stop the newly started `immich-server.service` and
+`immich-machine-learning.service` units.
+
+```console
+# systemctl stop immich-server.service immich-machine-learning.service
+```
+
+### Create database dump {#module-immich-migrating-from-docker-compose-dump-database}
+
+You might have done this already during your
+[backup step](#module-immich-migrating-from-docker-compose-backup).
+You can choose to reuse that database dump by making a copy, or just
+create another dump like this:
+
+```console
+# docker exec immich_postgres pg_dump --username immich immich -f /tmp/immich.sql
+# docker cp immich_postgres:/tmp/immich.sql ./immich.sql
+```
+
+::: {.note}
+While `pg_dump` offers a few output formats, this guide relies on it
+being `plain` (default), as we will have to modify it later.
+:::
+
+### Fix up database dump {#module-immich-migrating-from-docker-compose-fixup-dump}
+
+The database dump generated by `pg_dump`, can most likely not be
+imported as is. There are two changes we need to make, to allow it to
+be imported correctly.
+
+First, we need to fix the `search_path` variable of the dump, as a bug
+in PostgreSQL's `earthdistance` extension seems to cause issues when
+importing database dumps that contain its units.
+(See [this StackOverflow comment](https://stackoverflow.com/questions/49702964/postgres-import-to-rds-could-not-execute-query-error-type-earth-does-not-ex#comment86441215_49702964))
+To work around this issue, we can adjust the `search_path` variable
+in the database dump. This variable is set within the first 20
+lines of your database dump, so you can just open up your database
+dump in your preferred text editor and apply the following changes:
+
+```diff
+--- a/immich.sql
++++ b/immich.sql
+@@ -10,7 +10,7 @@ SET lock_timeout = 0;
+ SET idle_in_transaction_session_timeout = 0;
+ SET client_encoding = 'UTF8';
+ SET standard_conforming_strings = on;
+-SELECT pg_catalog.set_config('search_path', '', false);
++SELECT pg_catalog.set_config('search_path', 'public', false);
```

Comment:
The immich documentation at https://immich.app/docs/administration/backup-and-restore also adds `pg_catalog`.
```suggestion
+SELECT pg_catalog.set_config('search_path', 'public, pg_catalog', false);
```